### ISSUE TITLE
How do I fully uninstall?

### ISSUE THREAD

**Issue by Pomax** [sandboxie-plus/Sandboxie#1036]
**Describe the bug**
I tried to uninstall sandboxie plus. Windows reports it's done, but after rebooting the sandboxie icon still shows in the systray, even though the service no longer exists. `C:\Program Files\Sandboxie` _does_ still exist: how do I force-uninstall everything relating to sandboxie? Because the msi installer for some reason doesn't have an "uninstall" option baked in, like good installers (offering install, repair, and uninstall, from the same .msi file)

**System details and installed software (please provide the following information):**
 - What is your Windows edition and version? Win 10 pro x64 v2004 (build 19041.1110)
 - What is your current Sandboxie edition and version? v0.8.8
 - Please mention in which version this bug got introduced: no idea
 - Please mention any security software running in the background: standard windows defender
 -

**Comment by isaak654** [comment]
> Because the msi installer for some reason doesn't have an "uninstall" option baked in, like good installers (offering install, repair, and uninstall, from the same .msi file)

Question: which msi installer, exactly? I can only see .exe installers:

![0.8.8](https://user-images.githubusercontent.com/12372772/126884682-072e1edd-e91f-4059-92b2-429dc1ec0871.png)

**Comment by isaak654** [comment]
In case you upgraded from a very old version , see also #970.
That's happened to me after restoring an old backup with a very old Plus installation by mistake.

**Comment by Pomax** [comment]
Deleting files doesn't clean up a registry or startup setting etc. though? It would probably solve the problem but only by now having non-working but active entries in the registry that would reactivate if something figured out it could just rebuild those files for hilarious side effects =(

**Comment by DavidXanatos** [comment]
isaak654 asks a good question what msi installer? we don't have MSI installers yet.

In case you mean the sandboxie plus installer because it has a somewhat similar icon.

If something failed with the uninstall the best approach is to install again and than uninstall again.

**Comment by isaak654** [comment]
> 
> 
> Deleting files doesn't clean up a registry or startup setting etc. though? It would probably solve the problem but only by now having non-working but active entries in the registry that would reactivate if something figured out it could just rebuild those files for hilarious side effects =(

Could you mention the registry keys and files that stay as leftover for you?

You're not giving much till now, so I made a more accurate attempt (on my usual VM) with the recent 0.8.8 version.

1) Before my 0.8.8 uninstall, startup registry key is there:
![immagine](https://user-images.githubusercontent.com/12372772/126891827-c82a1ea6-7aa5-4c30-9413-ad49a4f2762b.png)
After my 0.8.8 uninstall, startup registry key is gone:
![immagine](https://user-images.githubusercontent.com/12372772/126891891-40fc5f87-b8d5-410c-9100-cf485d1840b1.png)

Unfortunately it doesn't seem the case with the context menu entry, after the uninstall + reboot:
![immagine](https://user-images.githubusercontent.com/12372772/126892101-da134139-4027-46f6-a367-043039df02c3.png)

2) My `C:\Program Files\Sandboxie-Plus` remains empty after the uninstall, probably because of the possibility to keep sandboxie.ini or other config stuff there.

3) AppData\Local\Sandboxie-Plus\Sandboxie-Plus.ini stays always there for future reinstallations

My considerations:
1) I would suggest to remove the context menu registry key above during the uninstall without exceptions
2) I would suggest one prompt before the uninstall process that asks "Do you want to wipe your configuration files (Sandboxie.ini and Sandboxie-Plus.ini)? Please remember to do a backup, just in case." That should clean also the Sandboxie-Plus folder in AppData\Local.

**Comment by Pomax** [comment]
That's because I don't _have_ any information to give you: I don't know what gets left behind, but given that sandboxie plus was still running, somehow, after uninstalling means that _something_ was clearly still left behind, that's why I asked if there was some way to automatically fully uninstall even after a regular uninstall attempt: I have no idea what's left and where, but stuff was clearly stlll left in several places. I've followed your advice your remove the entire folder, I then ended up doing several passes of CCleaner until there were no registry issues anymore (then uninstalled it because it's a giant piece of adware) and that seems to have done the trick.

Unfortunately, while reinstalling 0.8.8 then seemed to work, it wasn't able to run an installer sandboxed, where it wasn't writing any files to the sandbox (which at the same time making drive letters it was never going to need), so I guess I'm going to give windows sandbox a try. It's far of a hassle to use, but it seems to not run into "not v1.0 yet" issues at least.

Thanks for your work, and I hope Sophos goes out of business for how it treated sandboxie.

**Comment by isaak654** [comment]
> 
> Thanks for your work, and I hope Sophos goes out of business for how it treated sandboxie.

You're welcome. If you know someone that appreciates Sandboxie like we do, just spread the word... we need as many contributors/developers as possible in this repository.

> 
> Unfortunately, while reinstalling 0.8.8 then seemed to work, it wasn't able to run an installer sandboxed, where it wasn't writing any files to the sandbox (which at the same time making drive letters it was never going to need), so I guess I'm going to give windows sandbox a try. It's far of a hassle to use, but it seems to not run into "not v1.0 yet" issues at least.

Which is the installer in question?

**Comment by Pomax** [comment]
Fat Sounds' installer for their new "Boom Bap" product (a VST audio plugin). It's an exe installer that then extracts an msi file and runs that. But by the time it tries to run it, the sandbox dir has no files, it only has drive letters preallocated. The list of drive letters on the right is the result of hitting "show me all space occupied by subdirs", and there are none to list. Each "drive" is an empty folder.

![image](https://user-images.githubusercontent.com/177243/126907703-a3216639-f67e-4926-96f5-5b9ca20a38d1.png)

**Comment by Pomax** [comment]
There _is_ a `user` dir in parallel to the `drivers` dir, and that apparently has the content that the installer's looking for:

![image](https://user-images.githubusercontent.com/177243/126907727-19c459d6-2990-451a-a888-68287f34cb4e.png)

So... that dir should be `DefaultBox\drive\C\user`, not `DefaultBox\user`?

**Comment by isaak654** [comment]
Please try the workaround posted in #1007, and let me know.
The error is quite similar to your first screenshot, so it might fix your issue.

**Comment by Pomax** [comment]
How would I do that? Right now, I just right click the installer, and then use "run sandboxed".

**Comment by isaak654** [comment]
> 
> 
> How would I do that? Right now, I just right click the installer, and then use "run sandboxed".

Assuming you're on the main Sandboxie Plus window and you've closed any running program in the sandbox(es):
1) Go to Options menu -> Edit ini file -> confirm yes.
2) Add `MsiInstallerExemptions=y` below [DefaultBox], assuming that you want to run the program in DefaultBox afterwards.
3) Go to Option -> Reload ini file. You're ready to right click the installer, and then use "run sandboxed".

**Comment by isaak654** [comment]
> 
> 
> How would I do that? Right now, I just right click the installer, and then use "run sandboxed".

Assuming you're on the main Sandboxie Plus window and you've closed any running program in the sandbox(es):
1) Go to Options menu -> Edit ini file -> confirm yes.
2) Add `MsiInstallerExemptions=y` below [DefaultBox], if you want to run the program in DefaultBox afterwards.
3) Go to Options -> Reload ini file. You're ready to right click the installer, and then use "run sandboxed".

**Comment by Pomax** [comment]
Thank you - I've made those changes, reran the installer, and things look to be working now.

**Comment by isaak654** [comment]
> 
> 
> Thank you - I've made those changes, reran the installer, and things look to be working now. I'll keep an eye out for 0.9.0

You're welcome!
It would be advisable to remove the line `MsiInstallerExemptions=y` when you will upgrade to the next 0.8.9 or 0.9.0, this would help us to verify if the bug is definitely fixed on your side.

**Comment by Pomax** [comment]
If I remember, I will - however, it might also be a good idea to add a dialog box during the upgrade installer that goes "it looks like you customized your `ini` file and some of these settings are part of what changes. Here's the changelog, and here's your ini file: please verify you still want all of these, and hit save before this upgrade will continue".

I mean, that's obviously a bit wordy, but you get the idea. Point out that customizations related to the changelog need the user's explicit attention and the update won't finish until they do whatever is the right thing (with a little bit of text guidance).
